e2e/test/MethodE2EMultiplexingTests.cs

```diff
@@ -158,18 +244,32 @@ public async Task Method_IoTHubSak_DeviceReceivesMethodAndResponse_MuxWithPoolin
             };
 
             ICollection<DeviceClient> deviceClients = new List<DeviceClient>();
+            Dictionary<DeviceClient, int> deviceClientConnectionStatusChangeCount = new Dictionary<DeviceClient, int>();
 
             try
             {
                 for (int i = 0; i < devicesCount; i++)
                 {
-                    TestDevice testDevice = await TestDevice.GetTestDeviceAsync($"{DevicePrefix}_{i}_", type).ConfigureAwait(false);
+                    ConnectionStatus? lastConnectionStatus = null;
+                    ConnectionStatusChangeReason? lastConnectionStatusChangeReason = null;
+                    int setConnectionStatusChangesHandlerCount = 0;
+
+                    TestDevice testDevice = await TestDevice.GetTestDeviceAsync($"{DevicePrefix}_{i}_").ConfigureAwait(false);
                     DeviceClient deviceClient = testDevice.CreateDeviceClient(transportSettings, authScope);
                     deviceClients.Add(deviceClient);
 
+                    deviceClient.SetConnectionStatusChangesHandler((status, statusChangeReason) =>
+                    {
+                        setConnectionStatusChangesHandlerCount++;
```

Comment:
Same comments from the other tests.
Looks like this code is common: please extract the common pattern (use template method for example) and apply to all tests. (For an example see the FaultInjection tests.)